### PR TITLE
[4] remove Factory::getDbo() deprecated

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
get DatabaseDriver from container

### Testing Instructions
code review

### Actual result BEFORE applying this Pull Request
@deprecated 5.0 — Load the database service from the dependency injection container

### Expected result AFTER applying this Pull Request
no more `Factory::getDbo(); @deprecated 5.0 
`

### CODE REVIEW

**Comment by PhilETaylor** [comment]
I’m pretty sure there are now three open PRs all the same to do this

**Comment by alikon** [comment]
i've found only this #35888 ... but is not complete..what is the other one ?

File: libraries/src/Menu/SiteMenu.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function __construct($options = array())
 	{
 		// Extract the internal dependencies before calling the parent constructor since it calls $this->load()
 		$this->app      = isset($options['app']) && $options['app'] instanceof CMSApplication ? $options['app'] : Factory::getApplication();
-		$this->db       = isset($options['db']) && $options['db'] instanceof DatabaseDriver ? $options['db'] : Factory::getDbo();
+		$this->db       = isset($options['db']) && $options['db'] instanceof DatabaseDriver ? $options['db'] : Factory::getContainer()->get('DatabaseDriver');
```
Comment (richard67) [inline libraries/src/Menu/SiteMenu.php]:
```suggestion
		$this->db       = isset($options['db']) && $options['db'] instanceof DatabaseDriver
			? $options['db']
			: Factory::getContainer()->get('DatabaseDriver');
```
This should hopefully make PHPCS pass in drone.

**Comment by HLeithner** [comment]
it would be nice to add the variable type, so the static code analyser has a chance to help auto completing.

**Comment by alikon** [comment]
outside pr scope

**Comment by PhilETaylor** [comment]
> outside pr scope

I disagree. The code you are replacing has a defined return type in the docblock which is used by static analysis, whereas after this PR it's not defined.

**Comment by alikon** [comment]
ok better status quo then

**Comment by PhilETaylor** [comment]
Wow attitude much - just add the 12 lines of copy and paste doc code and move on

**Comment by PhilETaylor** [comment]
Replacement PR Provided as #36511

**Pull request closed without merging**